### DISCUSSION TITLE
Package Rules `target:extraconf`

### DISCUSSION THREAD

**Discussion by qudix** [xmake-io/xmake#2947] [Q&A]
When using the new package rules feature in 2.7.2, how can a rule refer to itself when using `target:extraconf("rules", "???", "conf")`? 
Or is there an alternative?

**Comment by waruqi** [comment]
> When using the new package rules feature in 2.7.2, how can a rule refer to itself when using `target:extraconf("rules", "???", "conf")`? 

right, did you try it? or use `target:rule("@zlib/test")`

**Comment by qudix** [comment]
Using the form `@package/rule` does work, it just seems odd. I suppose it's very unlikely for the package name to change so it's not a big deal.
